back/spec/services/local_project_copy_service_spec.rb

```diff
@@ -78,79 +88,136 @@
+        end)
+
+      expect(copied_project.map_config.legend_items.map do |record|
+        record.as_json(except: %i[id map_config_id updated_at created_at])
+      end)
+        .to match_array(continuous_project.map_config.legend_items.map do |record|
+          record.as_json(except: %i[id map_config_id updated_at created_at])
+        end)
     end
 
     it 'copies associated volunteering_causes' do
-      cause = create(:cause, title_multiloc: { en: 'Test cause' })
-      source_project = Project.find(cause.participation_context_id)
-
-      copied_project = service.copy(source_project)
-      expect(copied_project.causes.first.participation_context_id).to eq copied_project.id
-      expect(copied_project.causes.first.as_json(except: %i[id participation_context_id image updated_at created_at]))
-        .to eq source_project.causes.first.as_json(except: %i[id participation_context_id image updated_at created_at])
+      create_list(:cause, 2, participation_context_id: continuous_project.id, participation_context_type: 'Project')
+      copied_project = service.copy(continuous_project.reload)
+
+      expect(copied_project.causes.map do |record|
+        record.as_json(except: %i[id participation_context_id image updated_at created_at])
+      end)
+        .to match_array(continuous_project.causes.map do |record|
+          record.as_json(except: %i[id participation_context_id image updated_at created_at])
+        end)
     end
 
     it 'copies associated custom_forms & related custom_fields & custom_field_options' do
-      custom_form = create(:custom_form)
-      custom_field = create(:custom_field, resource_type: 'CustomForm', input_type: 'select', resource_id: custom_form.id)
-      create(:custom_field_option, custom_field_id: custom_field.id)
-      source_project = custom_form.participation_context
+      custom_form = create(
+        :custom_form,
+        participation_context_id: continuous_project.id,
+        participation_context_type: 'Project'
+      )
+      create_list(:custom_field_select, 5, :with_options, resource_type: 'CustomForm', resource_id: custom_form.id)
+      copied_project = service.copy(continuous_project)
+
+      expect(copied_project.custom_form.custom_fields.map do |record|
+        record.as_json(except: %i[id ordering resource_id updated_at created_at])
+      end)
+        .to match_array(continuous_project.custom_form.custom_fields.map do |record|
+          record.as_json(except: %i[id ordering resource_id updated_at created_at])
+        end)
+
+      # Test ordering values can be used to order copied custom_fields in same order as source records,
+      # even when the ordering values are not exact copies of the integer values in the source custom_fields.
+      expect(continuous_project.custom_form.custom_fields.order(:ordering).pluck(:key))
+        .to eq(copied_project.custom_form.custom_fields.order(:ordering).pluck(:key))
```

Comment:
This esoteric take on testing ordering might well not be necessary once issues with ordering are addressed in the Epic Sebastien mentioned in this [Slack thread](https://citizenlabco.slack.com/archives/C016C2EHURY/p1674555768857019).